File: Sources/Features/DragonBall/DragonBallViewModel.swift
import SwiftUI

struct DBCharacter: Identifiable {
    let id: String
    let name: String
    let race: String
    let voiceActor: String
}

class DragonBallViewModel: ObservableObject {
    @Published var characters: [DBCharacter] = []
    
    init() {
        characters = [
            DBCharacter(id: "1", name: "Goku", race: "Saiyan", voiceActor: "Sean Schemmel"),
            DBCharacter(id: "2", name: "Vegeta", race: "Saiyan", voiceActor: "Christopher Sabat"),
            DBCharacter(id: "3", name: "Piccolo", race: "Namekian", voiceActor: "Christopher Sabat"),
            DBCharacter(id: "4", name: "Gohan", race: "Half-Saiyan", voiceActor: "Kyle Hebert"),
            DBCharacter(id: "5", name: "Frieza", race: "Frieza Race", voiceActor: "Daman Mills")
        ]
    }
    
    func getCharacterById(id: String) -> DBCharacter? {
        characters.first { $0.id == id }
    }
}

File: feature/dragon_ball/src/main/java/com/example/lego/feature/dragonball/DragonBallViewModel.kt
package com.example.lego.feature.dragonball

import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

data class DBCharacter(
    val id: String,
    val name: String,
    val race: String,
    val voiceActor: String
)

@HiltViewModel
class DragonBallViewModel @Inject constructor() : ViewModel() {

    private val _characters = MutableStateFlow<List<DBCharacter>>(emptyList())
    val characters: StateFlow<List<DBCharacter>> = _characters

    init {
        _characters.value = listOf(
            DBCharacter("1", "Goku", "Saiyan", "Sean Schemmel"),
            DBCharacter("2", "Vegeta", "Saiyan", "Christopher Sabat"),
            DBCharacter("3", "Piccolo", "Namekian", "Christopher Sabat"),
            DBCharacter("4", "Gohan", "Half-Saiyan", "Kyle Hebert"),
            DBCharacter("5", "Frieza", "Frieza Race", "Daman Mills")
        )
    }

    fun getCharacterById(id: String): DBCharacter? {
        return _characters.value.find { it.id == id }
    }
}
